django/core/management/__init__.py

```diff
@@ -120,6 +120,14 @@ def call_command(command_name, *args, **options):
     arg_options = {opt_mapping.get(key, key): value for key, value in options.items()}
     defaults = parser.parse_args(args=[force_text(a) for a in args])
     defaults = dict(defaults._get_kwargs(), **arg_options)
+    # Check if any unrecognized options were passed.
+    private_options = command.base_private_options + command.private_options
+    recognized_options = {action.dest for action in parser._actions} | set(private_options)
+    unrecognized_options = set(options.keys()) - recognized_options
+    if unrecognized_options:
+        raise TypeError("Unrecognized %s: %r." % (
```

Comment:
Maybe something like "call_command() received unrecognized option(s) for the <foo> command: .... "
I think listing all the options in the message might not be a bad idea either if it doesn't look too cluttered.